main.py

```diff
@@ -0,0 +1,96 @@
+import random
+import re
+
+def get_user_inputs():
+    """Ask user for inputs."""
+    name_list = ''
+    while True:
+        # Ask the user to input a list of names separated by commas
+        name_list = input("Enter names separated by a comma: ")
+        #  Check the list of names is not empty
+        if not check_name_list_not_empty(name_list):
+            print("Error: You did not submit any names. Please enter a list of names separated by a comma: ")
+            continue
+        # Remove whitespace after each comma in the input
+        name_list = remove_whitespace_after_commas(name_list)
+        # Check that the input contains only letters, hyphens, and commas
+        if not check_punctuation(name_list):
+            print("Error: Name list should only contain letters, hyphens, and commas.")
+            continue
+        # Remove duplicates from the input list of names
+        name_list = remove_duplicates(name_list.split(','))
+        # Check for and remove any empty strings in the input list
+        name_list = check_for_empty_string(name_list)
+        break
+    return name_list
+
+def get_valid_num_of_names(name_list):
+    """Get a user input for the number that is an integer and lower than the number of names in the list submitted."""
+    while True:
+        # Ask the user to input a number lower than the number of names in the list
+        num_of_names = input(f"Enter a number {len(name_list)} or lower to randomly select that number of names from the list: ")
```

Comment:
Why not len(name_list) - 1?

Reply:
The length is the number of names in the list.  -1 would be wrong?